### DISCUSSION TITLE
How can I apply a bootstrap theme?

### DISCUSSION THREAD

**Discussion by Rieleniel** [alejandroautalan/pygubu-designer#262] [Q&A] [answered]
I am trying to use one of the predefined bootstrap themes in my application. I might be a noob, but can you show me an example of how to use it? Where do I go from here?

```self.presenter = Presenter()
self.builder = builder = pygubu.Builder()
self.themebuilder  = pygubu.theming.bootstrap.builder.BootstrapThemeBuilder("pbs_vapor")

builder.add_resource_path(GUI_PATH)
builder.add_from_file(GUI)```

**Comment by alejandroautalan** [comment, marked as answer]
Hello, @Rieleniel thanks for trying pygubu.

Sorry, I forgot to create an example for bootstrap themes. I will create one ASAP.

In the meantime, do this:

- Go to project settings and enable "Use ttk style definitions file", even if you are not using one.

This will generate the init constructor like:

```python
    def __init__(self, master=None, on_first_object_cb=None):
        super().__init__(master, on_first_object_cb=None)
```

- Import Style object from pygubu themes.

```python
from pygubu.theming.bootstrap.style import Style
```

- Create a method that will be called when the first window is created by the builder. And in this function setup the theme you want to use.

Example:

```python
#!/usr/bin/python3
import pathlib
import tkinter as tk
import pygubu
from pbsthemeui import ThemeTestUI

from pygubu.theming.bootstrap.style import Style


class ThemeTest(ThemeTestUI):
    def __init__(self, master=None, on_first_object_cb=None):
        super().__init__(
            master, 
            on_first_object_cb=self.setup_theme
        )
    
    def setup_theme(self, master):
        s = Style(master)
        s.theme_use("pbs_cyborg")


if __name__ == "__main__":
    app = ThemeTest()
    app.run()

```

Let me know if this helps you.
Regards
Alejandro A.

**Reply by Rieleniel** [reply]
Thanks! Based on this, I could implement the style I wanted. To paraphrase, I implemented it like so:

```python
class ThemeTestUI:

    def __init__(self, master = None) -> None:

        self.presenter = Presenter()
        self.builder = builder = pygubu.Builder()

        builder.add_resource_path(GUI_PATH)
        builder.add_from_file(GUI)
        self.mainwindow = builder.get_object('toplevel', master)

    def run(self):
            s = Style(self.mainwindow)
            s.theme_use("pbs_vapor")
            self.mainwindow.mainloop()
```

I didn't expect it to affect the tkinter.filedialog functions, but I was delighted to see it did! Thank you for your prompt and helpful response!

I'm looking forward to extensively annoy my friends and colleagues with various bootstrap themes in the future!
